tests/unit/restapi/v1/test_tags.py

```diff
@@ -0,0 +1,413 @@
+      during registration.
+    """
+    tag1_expected = registered_tags["tag1"]
+    tag2_expected = registered_tags["tag2"]
+    tag_expected_list = [tag1_expected, tag2_expected]
+
+    assert_retrieving_all_tags_works(
+        client,
+        expected=tag_expected_list,
+        search="name:*ML*",
+    )
+
+
+@pytest.mark.v1
+def test_cannot_register_existing_tag_name(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_tags: dict[str, Any],
+) -> None:
+    """Test that registering a tag with an existing name fails.
+
+    This test validates the following sequence of actions:
+
+    - A user registers a tag named "tag1".
+    - The user attempts to register a second tag with the same name, which fails.
+    """
+    existing_tag = registered_tags["tag1"]
+
+    assert_registering_existing_tag_name_fails(
+        client, name=existing_tag["name"], group_id=existing_tag["group_id"]
+    )
+
+
+@pytest.mark.v1
+def test_rename_tag(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_tags: dict[str, Any],
+) -> None:
+    """Test that a tag can be renamed.
+
+    This test validates the following sequence of actions:
+
+    - A user registers a tag named "old_name".
+    - The user is able to retrieve information about the "old_name" tag that
+      matches the information that was provided during registration.
+    - The user renames this same tag to "new_name".
+    - The user retrieves information about the same tag and it reflects the name
+      change.
+    """
+    tag1 = registered_tags["tag1"]
+    new_name = "new_name"
+
+    rename_tag(client, tag_id=tag1["tagId"], new_name=new_name)
+    assert_tag_name_matches_expected_name(
+        client, tag_id=tag1["tagId"], expected_name=new_name
+    )
+
+    tag2 = registered_tags["tag2"]
+
+    assert_cannot_rename_tag_with_existing_name(
+        client, tag_id=tag1["tagId"], existing_name=tag2["name"]
+    )
+
+
+@pytest.mark.v1
+def test_delete_tag_by_id(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+    registered_tags: dict[str, Any],
+) -> None:
+    """Test that a tag can be deleted by referencing its id.
+
+    This test validates the following sequence of actions:
+
+    - A user registers a tag.
+    - The user is able to retrieve information about the tag that
+      matches the information that was provided during registration.
+    - The user deletes the tag by referencing its id.
+    - The user attempts to retrieve information about the tag, which
+      is no longer found.
+    """
+    tag_expected = registered_tags["tag1"]
+
+    assert_retrieving_tag_by_id_works(
+        client, tag_id=tag_expected["tagId"], expected=tag_expected
+    )
+    delete_tag(client, tag_id=tag_expected["tagId"])
+
+    assert_tag_is_not_found(client, tag_id=tag_expected["tagId"])
```

Comment:
Same as other comment, I think you access the tag identifier with `id` instead of `tagId`:

```suggestion
    assert_retrieving_tag_by_id_works(
        client, tag_id=tag_expected["id"], expected=tag_expected
    )
    delete_tag(client, tag_id=tag_expected["id"])

    assert_tag_is_not_found(client, tag_id=tag_expected["id"])
```